components/Header.js

```diff
@@ -1,29 +1,48 @@
 import Logo from "./Logo";
 import Image from "next/image";
 import styled from "styled-components";
+import Navigation from "./Navigation";
+import { useState } from "react";
 
 export default function Header() {
+  const [hamburgerOpen, setHamburgerOpen] = useState(false);
+  function handleNavigation() {
+    setHamburgerOpen(!hamburgerOpen);
+  }
   return (
-    <StyledNav>
-      <StyledWrapper>
-        <Logo />
-        <StyledHeadline>Dog Genie</StyledHeadline>
-      </StyledWrapper>
-    </StyledNav>
+    <StyledSection>
+      {hamburgerOpen && <Navigation handleNavigation={handleNavigation} />}
+      <StyledHeader>
+        <StyledWrapper>
+          <Logo />
+          <StyledHeadline>Dog Genie</StyledHeadline>
+          <StyledHamburgerButton type="button" onClick={handleNavigation}>
+            <StyledHamburgerIcon
+              src="/hamburger-icon.svg"
+              alt="Icon of a hamburger menu"
+              width={40}
+              height={40}
+              blurDataURL="data:..."
+              placeholder="blur" // Optional blur-up while loading
+            />
+          </StyledHamburgerButton>
+        </StyledWrapper>
+      </StyledHeader>
+    </StyledSection>
   );
 }
 
 const StyledHeadline = styled.h2`
   font-size: 2rem;
 `;
 
-const StyledNav = styled.nav`
+const StyledHeader = styled.nav`
```

Comment:
The name suggests this is a header, but the element you are styling is a nav element.

```suggestion
const StyledHeader = styled.header`
```